send.go

```diff
@@ -189,5 +190,47 @@ func (c *Connection) send(ctx context.Context, request *http.Request) (response
 		}
 	}
 
+	// check if json
+	if !strings.EqualFold(response.Header.Get("Content-Type"), "application/json") {
+		err = fmt.Errorf("expected JSON content type but received '%s'; %s",
+			response.Header.Get("Content-Type"),
+			getResponseInfo(response))
+		return
+	}
+
+	// check if (maybe) empty
+	if response.Header.Get("Content-Length") == "0" {
+		err = fmt.Errorf("response is empty; %s", getResponseInfo(response))
+		return
+	}
```

Comment:
An empty response body is a perfectly valid response. In particular it is the expected response body for `DELETE`.

Reply:
makes sense to me. I removed.